### ISSUE TITLE
Cannot rename a SageMaker Endpoint Configuration

### ISSUE THREAD

**Issue by izeau** [hashicorp/terraform-provider-aws#21811]
### Community Note

* Please vote on this issue by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original issue to help the community and maintainers prioritize this request
* Please do not leave "+1" or other comments that do not add relevant new information or questions, they generate extra noise for issue followers and do not help prioritize the request
* If you are interested in working on this issue or have submitted a pull request, please leave a comment

### Terraform CLI and Terraform AWS Provider Version

* Terraform CLI 1.0.11
* Terraform AWS Provider 3.65.0

### Affected Resource(s)

* aws_sagemaker_endpoint_configuration
* aws_sagemaker_endpoint

### Terraform Configuration Files

Please include all Terraform configurations required to reproduce the bug. Bug reports without a functional reproduction may be closed without investigation.

```hcl
resource "aws_sagemaker_model" "model" {
  execution_role_arn = "arn:aws:..."
}

resource "aws_sagemaker_endpoint_configuration" "endpoint_config" {
  name = "endpoint-config"

  production_variants {
    variant_name           = "variant"
    model_name             = aws_sagemaker_model.model.name
    initial_instance_count = 1
    instance_type          = "ml.m5.large"
  }
}

resource "aws_sagemaker_endpoint" "endpoint" {
  name                 = "endpoint"
  endpoint_config_name = aws_sagemaker_endpoint_configuration.endpoint_config.name
}
```

### Debug Output

_Hard to obtain atm (running Terraform in a custom CI pipeline), will update if needed_

### Expected Behavior

Not really sure, `name_prefix` instead of `name` would be nice I guess

### Actual Behavior

1. Terraform destroys the old endpoint config
2. Terraform re-creates a new endpoint config
3. Terraform tries to update the endpoint
4. `ValidationException: Could not find endpoint configuration "<old_endpoint_config_arn>".`

### Steps to Reproduce

1. `terraform apply`
2. update the `aws_sagemaker_endpoint_configuration.endpoint_config.name` attribute
3. `terraform apply`

### Important Factoids

Nothing really atypical, although the SageMaker docs warns against deleting an endpoint configuration in use by an endpoint:

> **Note**
> You must not delete an `EndpointConfig` in use by an endpoint that is live or while the `UpdateEndpoint` or `CreateEndpoint` operations are being performed on the endpoint. To update an endpoint, you must create a new `EndpointConfig`.
>
> If you delete the `EndpointConfig` of an endpoint that is active or being created or updated you may lose visibility into the instance type the endpoint is using. The endpoint must be deleted in order to stop incurring charges. 
>
> – [UpdateEndpoint docs](https://docs.aws.amazon.com/sagemaker/latest/APIReference/API_UpdateEndpoint.html)

**Comment by izeau** [comment]
Found a workaround (kind of) in #14337: in addition to the autogenerated `name` for the `aws_sagemaker_endpoint_configuration` resource, we need to set the `create_before_destroy` lifecycle configuration:

```hcl
resource "aws_sagemaker_endpoint_configuration" "endpoint_config" {
  name = "endpoint-config"

  production_variants {
    variant_name           = "variant"
    model_name             = aws_sagemaker_model.model.name
    initial_instance_count = 1
    instance_type          = "ml.m5.large"
  }

  lifecycle {
    create_before_destroy = true
  }
}
```

I did not have to let Terraform generate the `aws_sagemaker_endpoint` name, though.

I still think this issue should be addressed, either in the docs or by adding a `name_prefix` attribute to the `aws_sagemaker_endpoint_configuration` resource. I would be glad to implement both changes if it is OK with you!

**Comment by Limess** [comment]
I agree adding a `name_prefix` would be useful here. The default terraform name is a pain to find in the UI.

Currently the other option seems to be to add something like a date generated by `time_static` as a suffix, but it's a bit painful to track all the changes which require the endpoint configuration to be recreated as it has a huge amount of configuration.
